orcid-web/src/main/webapp/static/javascript/ng1Orcid/app/shared/person.service.ts

```diff
@@ -1,30 +1,43 @@
 import { HttpClient } from "@angular/common/http";
-import { Injectable } from "@angular/core";
-import { Observable, BehaviorSubject } from "rxjs";
+import { Injectable, Component, OnDestroy } from "@angular/core";
+import { Observable, BehaviorSubject, Subject } from "rxjs";
+import { takeUntil } from 'rxjs/operators';
+import { CommonService } from './common.service.ts';
 
 @Injectable()
 export class PersonService {
+  private ngUnsubscribe: Subject<void> = new Subject<void>();
   private url: string;
   private path = "/person.json";
   private endpointWasCall: boolean = false;
   private personEndpoint;
   private response: BehaviorSubject<any> = new BehaviorSubject(null);
+  public userInfo: any;
 
-  constructor(private http: HttpClient) {
+  constructor(private http: HttpClient, private commonSrvc: CommonService) {
     this.url = getBaseUri();
+    this.userInfo = {};
   }
 
-  getPerson() {
-    if (!this.endpointWasCall) {
-      this.personEndpoint = this.http
-        .get(this.url + "/" + orcidVar.orcidId + this.path)
-        .subscribe(person => {
-          this.response.next(person);
-        });
-      this.endpointWasCall = true;
-      return this.response.asObservable();
+  getPerson() {      
+    if (!this.endpointWasCall) {        
+        this.commonSrvc.userInfo$.subscribe(
+                data => {
+                    this.userInfo = data;
+                    this.personEndpoint = this.http
+                    .get(this.url + '/' + this.userInfo['REAL_USER_ORCID'] + this.path)
```

Comment:
This might work fine. The problems is that two serial calls https (meaning the second one waiting for the first one) is going to add some time before the first contentful paint. 

On the prototype I am directly reading the url to quickly get the Orcid id

https://github.com/ORCID/orcid-angular/blob/master/src/app/profile/pages/profile-page/profile-page.component.ts#L25-L32

Reply:
Seems like you are doing something similar here https://github.com/ORCID/ORCID-Source/pull/5256/files#diff-19fa4feaac15184e044ab74e4e7ea3fdR73

Reply:
Hey @leomendoza123, the problem on the person.service.ts is that we don't have the iD in the url, because those are the pages for a logged in user: /my-orcid, /account, etc; so, not sure how to fix that, any suggestion?

Reply:
Hey @leomendoza123,

I created a new endpoint for logged in users: /person.json

so: /<ORCID>/person.json will fetch the data in the public page while /person.json will fetch the data when the user is logged in.

Please take another look.

Thanks